CONTRIBUTING.md

```diff
@@ -18,12 +18,26 @@ pip install -e ".[dev]"
 
 &nbsp;
 
-## Unit Tests and Recipe Tests
-For running unit tests locally:
-```pytest tests```
-
-For running recipe tests locally (requires access to private S3 bucket):
-```./tests/recipes/run_test.sh```
+## Testing
+TorchTune contains three different types of tests: unit tests, recipe test, and regression tests. These tests are distinguished by their complexity and the resources they require to run. Recipe tests and regression tests are explicitly marked via pytest.mark decorators.
+
+- **Unit tests**
+  - These should be minimal tests runnable without remote access. (No large models, no downloading weights). Unit tests should be under [tests/torchtune](https://github.com/pytorch/torchtune/tree/main/tests/torchtune).
+  - All unit tests can be run via ```pytest tests```.
+- **Recipe tests**
+  - These are relatively small-scale integration tests for running our recipes. These include
+  both single-device recipes and distributed recipes. In the latter case, tests should be marked with the `@gpu_test` decorator to indicate how many GPUs they need to run.
+  - Recipe tests require remote access as (small) model weights will be downloaded from S3 to run them.
```

Comment:
nit: but would change s3 to hf hub cause that's likely what most of our users will be using with the exception of the core torchtune team

Reply:
Oh interesting. My assumption here was that we would always use S3 for testing assets. So do you envision us just pushing our toy models to the hub and having some kind of fallback S3 -> HF hub based on permissions then?